### ISSUE TITLE
Interim Trial Licenses destroy custom schema data

### ISSUE THREAD

**Issue by dbryar** [pingidentity/terraform-provider-pingone#688]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### PingOne Terraform provider Version

0.18.1

### Terraform Version

1.4.2

### Affected Resource(s)

- pingone_schema

### Debug Output

```log
...
  # data.pingone_schema.user will be read during apply
  # (depends on a resource or a module with changes pending)
 <= data "pingone_schema" "user" {
      + description    = (known after apply)
      + environment_id = "cf4a****111aec"
      + id             = (known after apply)
      + name           = "User"
    }
...
  # pingone_environment.stage will be updated in-place
  ~ resource "pingone_environment" "stage" {
        id                    = "cf4a****111aec"
      ~ license_id            = "cb5a****8392bb" -> "ed81****baa1d4"
        name                  = "Stage"
...
  # pingone_schema_attribute.profile must be replaced
-/+ resource "pingone_schema_attribute" "profile" {
      ~ id             = "6e9837ab-9fdf-4c2f-9877-4e26871c6018" -> (known after apply)
      + ldap_attribute = (known after apply)
        name           = "profile"
      ~ required       = false -> (known after apply)
      ~ schema_id      = "d7fa113b-8984-4958-bf86-47af16626816" # forces replacement -> (known after apply)
      ~ schema_type    = "CUSTOM" -> (known after apply)
        # (7 unchanged attributes hidden)
    }
...
```
and 7 more custom schema elements

### Expected Behaviour
Should apply existing state without overwriting/replacing schema attributes
Should not delete schema data

### Actual Behaviour
Overwrites schema, deleting ALL custom attribute data on ALL users

### Steps to Reproduce
1. create `.tf` using a production license, with custom user schema elements 
2. `terraform apply`
3. add custom schema values to a user
4. change the license to a trial license (with custom schema enabled)
5. `terraform apply`
6. ALL users with custom schema data have had the custom values removed 

### Important Factoids
Due to additional features being added to the enterprise licence negotiations have led to an overrun on the expiration date, and a matching trial license was supplied as an interim measure. Using this license has annihilated the custom schema values across the user base

Permanent data loss is the result.

**Comment by patrickcping** [comment]
Hi @dbryar thanks for raising - this was previously identified in #330 and fixed in version `v0.19.0` (released on 8th August, current latest version is `v0.24.0`).  Agreed it can lead to data loss, and as `v0.24.0` contains a rollup of a number of important fixes, the advice is to maintain upgrades to the latest provider version to ensure stability in configuration pipelines.

When upgrading, you'll see that `pingone_schema_attribute.schema_id` is now deprecated and has been made optional, you would need to remove that parameter from the resource definition to ensure the fix.

If you're not able to upgrade the provider version, you can use the `lifecycle.ignore_changes` meta argument to tell Terraform not to recalculate the `pingone_schema_attribute` based on perceived upstream changes of the `pingone_schema_attribute.schema_id` parameter.

Additional context can found in issue #330 

As already resolved, will close as a duplicate

**Comment by dbryar** [comment]
```hcl
resource "pingone_schema_attribute" "profile" {
  environment_id = pingone_environment.dev.id
  name           = "profile"
  display_name   = "Profile ID"
  description    = "Profile ID in the Beem user database"
  type           = "STRING"
  unique         = false
  enabled        = true
  multivalued    = false
  lifecycle = {
    ignore_changes = true
  }
}
```

Results in 
```log
│ Error: Unsupported argument
│ 
│   on dev.tf line 711, in resource "pingone_schema_attribute" "profile":
│  711:   lifecycle = {
│ 
│ An argument named "lifecycle" is not expected here.
```
